inc/Ui/Diff.php

```diff
@@ -2,502 +2,190 @@
-        //dropdown
-        $form = new Form(['action' => wl()]);
-        $form->setHiddenField('id', $ID);
-        $form->setHiddenField('difftype', $this->difftype);
-        $form->setHiddenField('rev2[1]', $r_rev);
-        $form->setHiddenField('do', 'diff');
-        $input = $form->addDropdown('rev2[0]', $l_revisions)->val($l_rev)->addClass('quickselect');
-        $input->useInput(false); // inhibit prefillInput() during toHTML() process
-        $form->addButton('do[diff]', 'Go')->attr('type','submit');
-        $l_nav .= $form->toHTML();
-        //move forward
-        if ($l_next && ($l_next < $r_rev || !$r_rev)) {
-            $l_nav .= $this->diffViewlink('diffnextrev', $l_next, $r_rev);
-        }
 
-        /*
-         * Right side:
-         */
-        $r_nav = '';
-        //move back
-        if ($l_rev < $r_prev) {
-            $r_nav .= $this->diffViewlink('diffprevrev', $l_rev, $r_prev);
-        }
-        //dropdown
-        $form = new Form(['action' => wl()]);
-        $form->setHiddenField('id', $ID);
-        $form->setHiddenField('rev2[0]', $l_rev);
-        $form->setHiddenField('difftype', $this->difftype);
-        $form->setHiddenField('do', 'diff');
-        $input = $form->addDropdown('rev2[1]', $r_revisions)->val($r_rev)->addClass('quickselect');
-        $input->useInput(false); // inhibit prefillInput() during toHTML() process
-        $form->addButton('do[diff]', 'Go')->attr('type','submit');
-        $r_nav .= $form->toHTML();
-        //move forward
-        if ($r_next) {
-            if ($pagelog->isCurrentRevision($r_next)) {
-                //last revision is diff with current page
-                $r_nav .= $this->diffViewlink('difflastrev', $l_rev);
-            } else {
-                $r_nav .= $this->diffViewlink('diffnextrev', $l_rev, $r_next);
-            }
-        } else {
-            $r_nav .= $this->diffViewlink('diffbothnextrev', $l_next, $r_next);
-        }
-        return array($l_nav, $r_nav);
+    /**
+     * Build header of diff HTML
+     *
+     * @param string $l_rev   Left revisions
+     * @param string $r_rev   Right revision
+     * @return string[] HTML snippets for diff header
+     * @deprecated 2020-12-31
+     */
+    public function buildDiffHead($l_rev, $r_rev)
+    {
+        dbg_deprecated('not used see '. \dokuwiki\Ui\PageDiff::class .'::show()');
     }
 
     /**
-     * Create html link to a diff view defined by two revisions
+     * item url generator
      *
-     * @param string $linktype
-     * @param int $lrev oldest revision
-     * @param int $rrev newest revision or null for diff with current revision
-     * @return string html of link to a diff view
+     * @param string $id  page id or media id
+     * @param string|array $urlParameters  URL parameters, associative array recommended
+     * @return string
      */
-    protected function diffViewlink($linktype, $lrev, $rrev = null)
+    protected function itemUrl($id, $urlParameters = '')
```

Comment:
Nice work, again!
I’m not expert in these kind of things, but could you not create typical an abstract `itemUrl()` and `itemFN()` here?